services/core/shared/core/transactionStatistics.js

```diff
@@ -249,6 +249,7 @@ const fetchTransactionsForPastNDays = async (n, forceReload = false) => {
 const init = async historyLengthDays => {
 	signals.get('blockIndexReady').add(() => {
 		fetchTransactionsForPastNDays(historyLengthDays);
+		signals.get('transactionStatsReady').dispatch(historyLengthDays);
```

Comment:
Consider making the callback function `async` and `await fetchTransactionsForPastNDays` for the signal dispatch to be accurate.